packages/block-library/src/cover/shared.js

```diff
@@ -35,7 +35,7 @@ export function dimRatioToClass( ratio ) {
 }
 
 export function attributesFromMedia( media ) {
-	if ( ! media || ! media.url ) {
+	if ( ! media || ( ! media.url && ! media.src ) ) {
```

Comment:
Oh, that's a good point! Yeah it would be nice. Media & Text seems to be slightly differently broken though 😅 (trying to drag an image into it actually removes the existing image instead of doing nothing) so I'd rather tackle that as a separate PR.